### ISSUE TITLE
libstableR doesn't build for macOS

### ISSUE THREAD

**Issue by swihart** [swihart/libstableR#1]
So the initial commit of this repo is the equivalent of  what is on CRAN, which as of  2017-07-25 02:48:50 is [failing for osx](https://cran.r-project.org/web/checks/check_results_libstableR.html):

<img width="571" alt="screen shot 2017-07-24 at 9 35 00 pm" src="https://user-images.githubusercontent.com/4581196/28551755-0653b426-70b8-11e7-978f-cc9b1e5e039a.png">

And indeed, when I do `devtools::install_github("swihart/libstableR")` at an R prompt I get the following:

```
> devtools::install_github("swihart/libstableR")
Downloading GitHub repo swihart/libstableR@master
from URL https://api.github.com/repos/swihart/libstableR/zipball/master
Installing libstableR
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet CMD  \
  INSTALL  \
  '/private/var/folders/tv/9xflkyxs21j3747s8wd641t1bd7ntf/T/RtmpndFVeN/devtoolsc82f77d9e3fa/swihart-libstableR-74696a3'  \
  --library='/Users/swihartbj/RLIB' --install-tests 

* installing *source* package ‘libstableR’ ...
** package ‘libstableR’ successfully unpacked and MD5 sums checked
** libs
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c RcppExports.cpp -o RcppExports.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c init.c -o init.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c mcculloch.c -o mcculloch.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c methods.c -o methods.o
clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_R.cpp -o stable_R.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_cdf.c -o stable_cdf.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_common.c -o stable_common.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_dist.c -o stable_dist.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_fit.c -o stable_fit.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_integration.c -o stable_integration.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_koutrouvelis.c -o stable_koutrouvelis.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_pdf.c -o stable_pdf.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_q.c -o stable_q.o
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I"/Users/swihartbj/RLIB/Rcpp/include" -I/usr/local/include   -fPIC  -Wall -g -O2  -c stable_rnd.c -o stable_rnd.o
clang++ -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o libstableR.so RcppExports.o init.o mcculloch.o methods.o stable_R.o stable_cdf.o stable_common.o stable_dist.o stable_fit.o stable_integration.o stable_koutrouvelis.o stable_pdf.o stable_q.o stable_rnd.o -lgsl -lgslcblas -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
installing to /Users/swihartbj/RLIB/libstableR/libs
** R
** demo
** inst
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
Error: package or namespace load failed for ‘libstableR’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/swihartbj/RLIB/libstableR/libs/libstableR.so':
  dlopen(/Users/swihartbj/RLIB/libstableR/libs/libstableR.so, 6): Symbol not found: _stable_set_THREADS
  Referenced from: /Users/swihartbj/RLIB/libstableR/libs/libstableR.so
  Expected in: flat namespace
 in /Users/swihartbj/RLIB/libstableR/libs/libstableR.so
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/swihartbj/RLIB/libstableR’
Installation failed: Command failed (1)
```

I believe  I have come up with a solution, that ultimately involves [two minor edits to
the DESCRIPTION and the src/Makevars](https://github.com/swihart/libstableR/commit/db0d26f03fd1ae40376133b0d230534db374fe6e) in addition to requiring the mac user to do the following:

1. be sure to have [homebrew](https://brew.sh/)

2. be sure to have [gsl](http://www2.lawrence.edu/fast/GREGGJ/CMSC210/gsl/gsl.html)

3. be sure to have Rcpp and [RcppGSL](https://cran.r-project.org/web/packages/RcppGSL/vignettes/RcppGSL-intro.pdf): just `install.packages("RcppGSL")` in R

Credits:

[Dirk Eddelbuettel answer on SO](https://stackoverflow.com/a/39153194/2727349)
[GREGGJ website on homebrew installing GSL on macs](http://www2.lawrence.edu/fast/GREGGJ/CMSC210/gsl/gsl.html)

**Comment by swihart** [comment]
Something here might help:  https://github.com/RcppCore/Rcpp/issues/723

**Comment by swihart** [comment]
After contacting authors, I'm going to post on SO to see if I can get help.  Doing the following changes https://github.com/swihart/libstableR/commit/db0d26f03fd1ae40376133b0d230534db374fe6e

Changes

```
** testing if installed package can be loaded
Error: package or namespace load failed for ‘libstableR’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/swihartbj/RLIB/libstableR/libs/libstableR.so':
  dlopen(/Users/swihartbj/RLIB/libstableR/libs/libstableR.so, 6): Symbol not found: _stable_set_THREADS
  Referenced from: /Users/swihartbj/RLIB/libstableR/libs/libstableR.so
  Expected in: flat namespace
 in /Users/swihartbj/RLIB/libstableR/libs/libstableR.so
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/swihartbj/RLIB/libstableR’
Installation failed: Command failed (1)
```
to

```
** testing if installed package can be loaded
Error: package or namespace load failed for ‘libstableR’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/swihartbj/RLIB/libstableR/libs/libstableR.so':
  dlopen(/Users/swihartbj/RLIB/libstableR/libs/libstableR.so, 6): Symbol not found: _libstableR_stable_cdf
  Referenced from: /Users/swihartbj/RLIB/libstableR/libs/libstableR.so
  Expected in: flat namespace
 in /Users/swihartbj/RLIB/libstableR/libs/libstableR.so
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/swihartbj/RLIB/libstableR’

Exited with status 1.

```

**Comment by swihart** [comment]
@hrbrmstr fixed with PR#1 https://github.com/swihart/libstableR/pull/2/commits/e5fb4cd0d0379de68be4da8072f2d6962048f2b1
